runtime/j9vm/j7vmi.c

```diff
@@ -2627,9 +2627,36 @@ jvmDefineClassHelper(JNIEnv *env, jobject classLoaderObject,
 																			J9_CP_INDEX_NONE,
 																			LOAD_LOCATION_UNKNOWN,
 																			NULL,
-																			NULL);
+																			NULL
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
+																			, NULL
+#endif /* defined(J9VM_OPT_VALHALLA_VALUE_TYPES) */
+																			);
 			/* Done if a class was found or and exception is pending, otherwise try to define the bytes */
 			if ((clazz != NULL) || (currentThread->currentException != NULL)) {
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
```

Comment:
you should check pending exception before doing another class load

Reply:
Fxied.